### ISSUE TITLE
[Ep] Support multiple versions of installs

### ISSUE THREAD

**Issue by ya55en** [ya55en/mashmallow-0#69]
[This is an "**Ep**ic" type of task which will be broken to stories; they might eventually be broken to tech tasks.]

We want to support multiple versions of applications installations.
This needs further discussion and a design description. For now, some concrete examples (these often appear as cases for a need of support for several versions):

- nodejs - different versions needed often;
- golang - different versions needed sometimes;
- gh cli - different versions may be needed (e.g. when the major version bumps as was the case recently);
- IDEs like PyCharm would be nice to have two versions in parallel for some time, while migrating to the new one.
